src/prefect/workers/base.py

```diff
@@ -1064,6 +1069,14 @@ def _emit_flow_run_executed_event(
             follows=submitted_event,
         )
 
+    async def _emit_worker_heartbeat_event(self, heartbeat_origin: str) -> Event:
+        return emit_event(
+            "prefect.worker.heartbeat",
+            resource=self._event_resource(),
+            related=self._event_related_resources(),
+            payload={"heartbeat_origin": heartbeat_origin},
```

Comment:
this SEEMS like a bespoke metadata dict, so i opted to pass in SOME context to this heartbeat event.  for now, it's just a string of the originating context, aka the class method where the event is being disaptched from

Reply:
Yep, that’s exactly what the payloads for.